lib/Util.php

```diff
@@ -65,31 +65,27 @@ public static function getTurnSettings(IConfig $config, $uid) {
 	 * Generates a username and password for the TURN server based on the
 	 *
 	 * @param IConfig $config
-	 * @param ISession $session
 	 * @param ITimeFactory $timeFactory
 	 * @return array
 	 */
-    public static function generateTurnSettings(IConfig $config, ISession $session, ITimeFactory $timeFactory) {
+    public static function generateTurnSettings(IConfig $config, ITimeFactory $timeFactory) {
 		// generate from shared secret
 		$turnServer = $config->getAppValue('spreed', 'turn_server', '');
 		$turnServerSecret = $config->getAppValue('spreed', 'turn_server_secret', '');
 		$turnServerProtocols = $config->getAppValue('spreed', 'turn_server_protocols', '');
-		$sessionId = $session->get('spreed-session');
 
-		if ($turnServer === '' || $turnServerSecret === '' || $turnServerProtocols === '' || empty($session)) {
+		if ($turnServer === '' || $turnServerSecret === '' || $turnServerProtocols === '') {
 			return array();
 		}
 
-		$time = $timeFactory->getTime();
-
 		// the credentials are valid for 24h - FIXME add the TTL to the response and properly reconnect then
-		$string = sprintf('%d:%s', $time + 86400, $sessionId);
-		$hashedString = hash_hmac('sha1', $string, $turnServerSecret, true);
+		$username = $timeFactory->getTime() + 86400;
```

Comment:
> Do not all users have a PHP session which could be used (hmaced or something)?

That can easily be changed by just deleting ones cookie. So if one really wants to get a ton of valid TURN access credentials they can still bypass this pretty easily.